pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -321,6 +348,15 @@ func (lb *ListenerBuilder) buildInboundChainConfigs() []inboundChainConfig {
 			if cc.bind == "" {
 				// If user didn't provide, pick one based on IP
 				cc.bind = getSidecarInboundBindIP(lb.node)
+				// if no global unicast address
+				if cc.bind == WildcardAddress || cc.bind == WildcardIPv6Address {
+					bindAddress, exBindAddresses := getDualStackActualWildcard(lb.node)
+					// if it's dual stack environment
+					if bindAddress != "" && exBindAddresses != "" {
+						cc.bind = bindAddress
+						cc.exBind = []string{exBindAddresses}
```

Comment:
IIRC before in dual stack we made 2 listeners. Now are we creating 2 listeners with 2 addresses each? Or did e remove the 2 listeners somewhere I missed?

Reply:
Hi @howardjohn, according to original solution, there are 2 listeners for a service with both IPv4 and IPv6 addresses. For current solution, there is only 1 listener and there are address and additional addresses (new field) for it. The former is for the first address of the service, and the latter is for the rest addresses of the service.